osu.Game/Screens/Play/Player.cs

```diff
@@ -82,15 +84,17 @@ private void load(AudioManager audio, BeatmapDatabase beatmaps, OsuConfigManager
                 try
                 {
                     // Try using the preferred user ruleset
-                    ruleset = osu == null ? Beatmap.BeatmapInfo.Ruleset.CreateInstance() : osu.Ruleset.Value.CreateInstance();
-                    HitRenderer = ruleset.CreateHitRendererWith(Beatmap);
+                    ruleset = osu == null ? Beatmap.BeatmapInfo.Ruleset : osu.Ruleset.Value;
                 }
                 catch (BeatmapInvalidForModeException)
                 {
                     // Default to the beatmap ruleset
-                    ruleset = Beatmap.BeatmapInfo.Ruleset.CreateInstance();
-                    HitRenderer = ruleset.CreateHitRendererWith(Beatmap);
+                    ruleset = Beatmap.BeatmapInfo.Ruleset;
                 }
+
+                rulesetInstance = ruleset.CreateInstance();
+
+                HitRenderer = rulesetInstance.CreateHitRendererWith(Beatmap);

```

Comment:
Nope, I did it this way because HitRenderer internally throws if it can't convert the beatmap.